### PR TITLE
[onert] Add shapeinf rules for Transpose

### PR DESCRIPTION
**Pull request by dr-venkman**
Add Shape inferencing rules for Transpose, accounting for dynamic tensors.

Signed-off-by: venkat.iyer venkat.iyer@samsung.com

### CODE REVIEW

File: runtime/onert/backend/cpu/ops/TransposeLayer.h [hunk 1]
```diff
@@ -44,6 +44,8 @@ class TransposeLayer : public ::onert::exec::IFunction
 
   void run();
 
+  void runSync() { run(); }
```
Comment (hyunsik-yoon) [inline runtime/onert/backend/cpu/ops/TransposeLayer.h:47]: Please check if this is in the scope of this PR. If not, please make another PR for this. (it's not a scope of shapeinf)

Comment (dr-venkman) [reply runtime/onert/backend/cpu/ops/TransposeLayer.h:47]: I probably got this in the process of resolving a merge conflict. I will revert the changes after the test results. Thanks for spotting.

File: runtime/onert/core/src/util/shapeinf/Transpose.cc [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+{
+  const auto input_idx{op.getInputs().at(ir::operation::Transpose::Input::INPUT)};
+  const auto &input = _operands.at(input_idx);
+
+  // get mutable output operand
+  const auto output_idx = op.getOutputs().at(0);
+  ir::Operand &output = _operands.at(output_idx);
+  const auto perm{op.param().perm};
+  // const auto rank{op.param().rank};
+  // if input is dynamic, or perm vector is not yet filled, output also becomes dynamic
```
Comment (hyunsik-yoon) [inline runtime/onert/core/src/util/shapeinf/Transpose.cc:55]: Could you check `perm vector is not yet filled` is correct comment? IMHO, our assumption here is that `perm` is always filled.

File: runtime/onert/core/src/util/shapeinf/Transpose.cc [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  output.info().shape(new_shape);
+}
+
+// DynamicInferer at execution time
+void DynamicInferer::visit(const ir::operation::Transpose &op)
+{
+  // check if output is not dynamic
+  auto output_ind = op.getOutputs().at(0);
+  auto output = _tensor_registry->getITensor(output_ind);
+  if (!output->is_dynamic())
```
Comment (hyunsik-yoon) [inline runtime/onert/core/src/util/shapeinf/Transpose.cc:72]: please check other PRs and fix this to `!input->is_dynamic()`

File: runtime/onert/core/src/util/shapeinf/Transpose.cc (hunk 2 above)
Comment (dr-venkman) [reply runtime/onert/core/src/util/shapeinf/Transpose.cc:55]: I will remove the comment line regarding `perm vector`. We do not check for it now, so the comment is out of place. Thanks.

File: runtime/onert/test/util/ShapeInference.cc [hunk 4]
```diff
@@ -312,3 +312,60 @@ TEST(ShapeInference, FullyConnected)
   ASSERT_EQ(infered_out_shape.dim(0), 36);
   ASSERT_EQ(infered_out_shape.dim(1), 3);
 }
+
+TEST(ShapeInference, Transpose)
+{
```
Comment (hyunsik-yoon) [inline runtime/onert/test/util/ShapeInference.cc:317]: :+1:

File: runtime/onert/core/src/util/shapeinf/Transpose.cc (hunk 3 above)
Comment (dr-venkman) [reply runtime/onert/core/src/util/shapeinf/Transpose.cc]: Modified condition check as suggested. Thanks for spotting.

**Comment by hyunsik-yoon** [comment]
Can I ask a favor of you?
Once anyone starts reviewing the PR, could you commit additional changes with `commit -m "short_message_without_signoff"`?
That way, reviewers can easily recognized what's been changed from the original commit. Otherwise, reviewers have to read the whole code repeatedly.

(please refer to an example: `Extend it to all for gradlew` commit in the middle of https://github.com/Samsung/ONE/pull/1776)

Thanks!

**Comment by dr-venkman** [comment]
> Can I ask a favor of you?
> Once anyone starts reviewing the PR, could you commit additional changes with `commit -m "short_message_without_signoff"`?
> That way, reviewers can easily recognized what's been changed from the original commit. Otherwise, reviewers have to read the whole code repeatedly.
> 
> (please refer to an example: `Extend it to all for gradlew` commit in the middle of #1776)
> 
> Thanks!

Noted, thanks for the reference pull request. Will keep in context.

**Comment by hyunsik-yoon** [comment]
Could you please add one of contributor in WG2? They know the logic of each kernel well. Thanks.

**Comment by dr-venkman** [comment]
> Could you please add one of contributor in WG2? They know the logic of each kernel well. Thanks.

Added @hseok-oh as reviewer. Thank you.
